### PR TITLE
[Feature] Low memory warning

### PR DESCRIPTION
**Pull request by annakocheshkova**
- Exposed the API to detect low memory warnings
- Added UI to trigger/display low memory warning
image

### CODE REVIEW

File: demoapp/config.xml [hunk 1]
```diff
@@ -29,13 +28,12 @@
...
+    <plugin name="cordova-plugin-whitelist" spec="^1.3.4" />
+    <plugin name="cordova-plugin-file" spec="^5.0.0" />
     <plugin name="cordova-plugin-camera" spec="^3.0.0" />
-    <plugin name="cordova-plugin-appcenter-analytics" spec="^0.3.6" />
-    <plugin name="cordova-plugin-appcenter-crashes" spec="^0.3.6" />
-    <plugin name="cordova-plugin-appcenter-push" spec="^0.3.6" />
-    <plugin name="cordova-plugin-appcenter-shared" spec="^0.3.6" />
-    <engine name="android" spec="^7.0.0" />
-    <engine name="ios" spec="5.0.0" />
+    <plugin name="cordova-plugin-appcenter-shared" spec="file:../cordova-plugin-appcenter-shared" />
```
Comment (annakocheshkova) [inline demoapp/config.xml]: Same here.
Please also be careful when updating basic plugins versions (like **file** plugin).
Changes like this should be tested cause they can produce some flaky behavior on cordova.
So I suppose the binary attachments to be tested after this.

File: demoapp/package.json [hunk 2]
```diff
@@ -14,18 +14,20 @@
...
     }
   },
   "dependencies": {
-    "cordova-android": "^7.0.0",
-    "cordova-ios": "5.0.0",
-    "cordova-plugin-appcenter-analytics": "^0.3.5",
-    "cordova-plugin-appcenter-crashes": "^0.3.5",
-    "cordova-plugin-appcenter-push": "^0.3.5",
-    "cordova-plugin-appcenter-shared": "^0.3.5",
+    "cordova-android": "^8.0.0",
```
Comment (annakocheshkova) [inline demoapp/package.json]: Same here. Is this tested?

File: cordova-plugin-generate-low-memory/src/ios/LowMemory.m [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- (instancetype) init {
+  self = [super init];
+  if (self) {
+    _buffers = [NSMutableArray new];
+    _allocated = 0;
+  }
+  return self;
+}
+
+- (void) generateLowMemory :(CDVInvokedUrlCommand *) command
```
Comment (jaelim-ms) [inline cordova-plugin-generate-low-memory/src/ios/LowMemory.m]: Format. No space after `(void)`.
Same for the above and below methods.

File: cordova-plugin-generate-low-memory/src/ios/LowMemory.m [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    memset(buffer, 42, blockSize);
+    [self.buffers addObject:[NSValue valueWithPointer:buffer]];
+    self.allocated += blockSize;
+    NSLog(@"Allocated %zu MB", self.allocated / (1024 * 1024));
+    [self produceLowMemory];
+  });
+}
+
++ (BOOL)requiresMainQueueSetup
+{
```
Comment (jaelim-ms) [inline cordova-plugin-generate-low-memory/src/ios/LowMemory.m]: `{` should be in the previous line.

File: cordova-plugin-generate-low-memory/src/ios/LowMemory.h [hunk 5]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+#import <Cordova/CDV.h>
+
+@interface LowMemory : CDVPlugin
+
+- (void) generateLowMemory: (CDVInvokedUrlCommand *)command;
```
Comment (jaelim-ms) [inline cordova-plugin-generate-low-memory/src/ios/LowMemory.h]: Format.

**Pull request merged**